server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -382,7 +384,7 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    public static void registerRequestHandler(TransportService transportService, SearchService searchService) {
```

Comment:
I think that you could leave this static and add the SearchTransportMetrics argument, not a huge deal but it helps following dependencies.

Reply:
I spent some time pulling on this thread; the changes are contained in this commit: https://github.com/elastic/elasticsearch/pull/102557/commits/40a878bef230a509fce67ef1033d6d1c8071287f